spark/src/main/scala/geotrellis/spark/io/cassandra/CassandraDriver.scala

```diff
@@ -0,0 +1,54 @@
+package geotrellis.spark.io.cassandra
+
+import java.nio.ByteBuffer;
+
+import geotrellis.spark._
+import geotrellis.raster._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import com.datastax.spark.connector.rdd.CassandraRDD
+import com.datastax.spark.connector._
+
+import com.datastax.driver.core.DataType.{text, blob}
+import com.datastax.driver.core.schemabuilder.SchemaBuilder
+import com.datastax.driver.core.Session
+
+class TableNotFoundError(table: String) extends Exception(s"Target Cassandra table '$table' does not exist.")
+
+trait CassandraDriver[K] extends Serializable {
+  def applyFilter(rdd: CassandraRDD[(String, ByteBuffer)], layerId: LayerId, filterSet: FilterSet[K]): RDD[(String, ByteBuffer)]
+  def decode(rdd: RDD[(String, ByteBuffer)], metaData: RasterMetaData): RasterRDD[K]
+  def rowId(id: LayerId, key: K): String
+
+  def load(sc: SparkContext, keyspace: String)(
+    id: LayerId, metaData: RasterMetaData, table: String, filters: FilterSet[K]): RasterRDD[K] = {
+
+    val rdd: CassandraRDD[(String, ByteBuffer)] = 
+      sc.cassandraTable[(String, ByteBuffer)](keyspace, table).select("id", "tile")
+
+    val filteredRDD = applyFilter(rdd, id, filters)
```

Comment:
Ah, I was under the impression that Spark does not fetch the tiles from Cassandra immediately. From [this guide](https://github.com/datastax/spark-cassandra-connector/blob/master/doc/3_selection.md#filtering-rows), using `where` instead of `filter` in the `applyFilter` method causes filtering to occur on the Cassandra side. I need to look into the connector and the `union` method more, but I believe Spark doesn't query for these tiles until either the union or some later point in time. 

Unfortunately with Cassandra, there's no built-in support for multi-range queries, so performing multiple 'where' queries is the only way to get that functionality.